Sources/libhostmgr/Model/RemoteVMRepository.swift

```diff
@@ -50,14 +56,25 @@ public struct RemoteVMRepository {
     }
 
     /// Downloads a remote image using atomic writes to avoid conflict with existing files or other processes
+    ///
+    @discardableResult
     public func download(
         image: RemoteVMImage,
-        to destination: URL,
-        progressCallback: FileTransferProgressCallback? = nil
-    ) async throws {
-        try await self.s3Manager.download(
+        progressCallback: @escaping FileTransferProgressCallback
+    ) async throws -> URL {
+
+        let destination = Configuration.shared.vmStorageDirectory
+
+        // Download the checksum file first
+        _ = try await self.s3Manager.download(
+            object: image.checksumObject,
+            to: destination.appendingPathComponent(image.checksumFileName),
+            progressCallback: nil
+        )
+
+        return try await self.s3Manager.download(
             object: image.imageObject,
-            to: destination,
+            to: destination.appendingPathComponent(image.fileName),
```

Comment:
I think previously S3 file was downloaded to a unique temporary image file, wasn't it? Now that we are downloading the image to a fixed location, there is chance the downloaded image file ends up corrupted, for example resuming a previous interrupted download. I'm not sure if Alamofire or URLSession has built-in support for resuming downloading, or will they throw an error if they can't resume the previous download (like the remote file has changed)?

Reply:
>Now that we are downloading the image to a fixed location

So this argument specifies where the file should go once it's fully downloaded. The `S3Manager.download` method handles the atomic copy (well...technically Alamofire does...) internally. This can be verified in the next PR, where you can kick off a VM download and see that there's no file in the destination path until the download completes, and you can see the file in the system tmp directory.

Reply:
Awesome, Alamofire is nice.